pkg/odo/cli/catalog/list/components.go

```diff
@@ -43,7 +43,11 @@ func (o *ListComponentsOptions) Complete(name string, cmd *cobra.Command, args [
 	o.Context = genericclioptions.NewContext(cmd)
 	o.catalogList, err = catalog.ListComponents(o.Client)
 	if err != nil {
-		return err
+		if experimental.IsExperimentalModeEnabled() {
+			log.Warning("Please log in to an OpenShift cluster to list OpenShift/s2i components")
```

Comment:
I have some concerns about _always_ displaying that warning message when on non-OpenShift Kube clusters (or with no Kube cluster).

Maybe this is more of question: Would it be better for us to just hide the error message if experimental mode is enabled? Or leave it as you have it?

Reply:
I agree with @johnmcollier I don't like always having that warning message.

Reply:
@cdrage @johnmcollier I totally agree with you guys, actually I think both ways are not perfect, and I think this is because by design we want to support both devfile and s2i together.

Always display warning message is annoying for devfile user since they don't actually need that message.

Hide warning message is not user friendly for s2i user since they still need the warning message for debugging purpose.

I think to make it better, I can use `glog` V logging style as John suggests so that devfile user won't see this warning message and s2i user can debug via -v flag.